app/controllers/account/admin/admin_controller.rb

```diff
@@ -1,2 +1,13 @@
 class Account::Admin::AdminController < Account::AccountController
+  def start_impersonate
+    user = User.where("id = ? AND role != ?", params[:user_id], 0).take
```

Comment:
create a scope non_admin in User model and just apply it here before searcing by id
don't use integers of roles 0 you specifically have :admin symbol for it